### PR TITLE
feat: Add form and question models

### LINKED ISSUE #17: Create back-end model classes for forms
## Description
As a developer,I need to create model classes for forms according to the database schema, so that I can use it for retrieving data.

## Mocks
db-structure-osp (2)

### Update [Required]
- [ ] create FormModel
- [ ] create QuestionAbstractModel

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description

- Added Form and Question models
- Added AbstractTimestamp model
- Added related Field models

Fixes #17 

### Type of Change:

- Code
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

The models are created and migrations run:
Screenshot from 2020-06-20 21-04-18

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented on my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: osp/admin.py [hunk 1]
```diff
@@ -3,5 +3,13 @@
 from osp.models import *
```
Comment (Monal5031) [inline osp/admin.py]: No! Please don't. Never do this

File: osp/models/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 from osp.models.user_information import UserInformation
-# from osp.models.form import Form
-# from osp.models.question import Question
\ No newline at end of file
+from osp.models.form import Form
+from osp.models.question import Question
+from osp.models.abstract_timestamp import AbstractTimestamp
+from osp.models.field import *
```
Comment (Monal5031) [inline osp/models/__init__.py]: Same here

File: osp/models/abstract_timestamp.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    created_on = models.DateTimeField(
+        auto_now_add=True
+    )
+    updated_on = models.DateTimeField(
+        auto_now=True
+    )
+
+    class Meta:
+
```
Comment (Monal5031) [inline osp/models/abstract_timestamp.py]: extra line, please remove

File: osp/models/abstract_timestamp.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from django.db import models
+
+class AbstractTimestamp(models.Model):
+    """
+    Abstract model for Created Time and Updated Time fields
+    """ 
+
+    created_on = models.DateTimeField(
```
Comment (Monal5031) [inline osp/models/abstract_timestamp.py]: single param, same line should suffice

File: osp/models/question.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+from django.db import models
+
+from osp.models.abstract_timestamp import AbstractTimestamp
+
+SA = 'char'
```
Comment (Monal5031) [inline osp/models/question.py]: Create a separate directory for utilities, This new dir should contain all the abstract or generic models, functions, choices, views and manager.

File: osp/models/question.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    (DD, 'Dropdown'),
+    (FI, 'File Upload'),
+    (DA, 'Date'),
+    (TI, 'Time')
+)
+
+class Question(AbstractTimestamp):
+
+    label = models.CharField(max_length=255)
+    data_type = models.CharField(
```
Comment (Monal5031) [inline osp/models/question.py:9]: Add a default as well

File: osp/models/form.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    )
+    target_user = models.CharField(
+        max_length=7,
+        choices=TARGET_USER,
+        default='all'
+    )
+
+    def __str__(self):
+
+        return f'{self.name}'
```
Comment (Monal5031) [inline osp/models/form.py]: F formatting for a single param doesn't makes sense

File: osp/models/form.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+class Form(AbstractTimestamp):
+
+    name = models.CharField(
+        max_length=255
+    )
+    description = models.TextField(
+        blank=True
+    )
+    created_on = models.DateTimeField(
```
Comment (Monal5031) [inline osp/models/form.py]: You are already inheriting `AbstractTimestamp`. no need to add these 2 fields again

File: osp/models/field.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
+from django.db import models
+from django.contrib.postgres.fields import ArrayField
+
+from osp.models.question import Question
+
+class Choice(Question):
```
Comment (Monal5031) [inline osp/models/field.py:6]: define __str__ for each of these, with their type of question

File: osp/models/question.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        max_length=8,
```
Comment (Monal5031) [inline osp/models/question.py:10]: Should be 16, `Paragraph Answer` is the longest

**Review by Monal5031 — CHANGES_REQUESTED** [review]
file line ending issues are in every file, make sure to check your IDE settings

**Review by sidvenu — CHANGES_REQUESTED** [review]
Please make the changes suggested by @Monal5031. I don't see any other changes.

File: osp/models/question.py (hunk 10 above)
Comment (bismitaguha) [reply osp/models/question.py:10]: It is taking the values of checkbox and dropdown to be the longest, that is, 8. Received this error in migrations, and was asked to keep maximum to be 8.

File: osp/models/form.py (hunk 7 above)
Comment (bismitaguha) [reply osp/models/form.py]: Without f the display is `{self.name}`. I feel it is needed!

**Review by Monal5031 — APPROVED** [review]

**Pull request merged**